integrationtests.py

```diff
@@ -719,11 +709,22 @@ def _buildAll(self):
             print("Starting compilation of {}".format(sourceFile))
             cxxflags = ["/c", "/nologo", "/EHsc"]
             cmd = CLCACHE_CMD + cxxflags + [sourceFile]
-            processes.append(subprocess.Popen(cmd))
+            processes.append(subprocess.Popen(cmd, env=self._createEnv()))
 
         for p in processes:
             p.wait()
 
+    def _enableNoDirect(self):
+        return False
```

Comment:
The implementations of this method never return a value which changes during the objects lifetime, so how about just passing the flag in the constructor and use it via a member variable?

In fact, since the variable is only used in `_createEnv`, how about passing a "base environment" in the constructor (and dropping the flag completely) and only making `_createEnv` add the `CLCACHE_DIR` variable if needed?